File: programme/random-odd-number/random-odd-number.java
// Write a programme to generate a random odd number in range n1-n2

import java.util.*;

class HelloWorld {

  public static void main(String[] args) {
    Scanner s_name = new Scanner(System.in);

    System.out.print("Enter start range : ");
    int startRange = s_name.nextInt();
    System.out.print("Enter end   range : ");
    int endRange = s_name.nextInt();

    System.out.print(
      "\nRandom odd value in int from " + startRange + " to " + endRange + " : "
    );
    int random_int = (int) Math.floor(
      Math.random() * (endRange - startRange + 1) + startRange
    );
    while (random_int % 2 == 0) {
      random_int =
        (int) Math.floor(
          Math.random() * (endRange - startRange + 1) + startRange
        );
    }
    System.out.println(random_int);

    s_name.close();
  }
}

File: programme/random-odd-number/random-odd-number.js
// Write a programme to generate a random odd number in range n1-n2

function randomOddNumber(n1, n2){
    let number = Math.floor(Math.random() * (n1 - n2 + 1)) + n2;

    while (number%2===0){
        number = Math.floor(Math.random() * (n1 - n2 + 1)) + n2;
    }

    return number;
}

const n1 = 0;
const n2 = 100;

console.log("Start Range: ", n1);
console.log("End Range:   ", n2);

console.log("Odd Number:  ", randomOddNumber(n1, n2));